[saajidshaik02] iP

saajidshaik02 opened the pull request:
# RizzBOT

> ““CS students can't get women. Let us help you!” – this guy probably [(source)](https://www.youtube.com/watch?v=2qBlE2-WL60)

RizzBOT frees your mind of having to remember your dates and things to do for yo girl. It’s,
- text-based
- easy to learn
- **FAST** _SUPER FAST_ to use

All you need to do is:
1. download it from [here](https://github.com/saajidshaik02/ip).
2. double-click it.
3. download and run app
4. add your tasks.
5. let it manage your tasks for you 😊

And it is **FREE**!

### Features:
- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
- [ ] Get women (hopefully)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

saajidshaik02 committed:
Rizz Level-0: Rename, Greet, Exit

saajidshaik02 committed:
Rizz Level-1: Echo

saajidshaik02 committed:
Duke Level-2: Add, List

saajidshaik02 committed:
Rizz Level-3: Mark as Done

saajidshaik02 committed:
Rizz Level-4: ToDo, Event, Deadline

saajidshaik02 committed:
Rizz A-TextUiTesting: Automated Text UI Testing

saajidshaik02 committed:
Rizz Level-5: Handle Errors

saajidshaik02 committed:
Rizz Level-6: Delete

saajidshaik02 committed:
Fix indentation

starchypotatocode commented on src/main/java/Rizz.java:
```diff
@@ -0,0 +1,154 @@
+        if (arrList.isEmpty()) {
+            System.out.println("\tNo items in the list.\n");
+        } else {
+            StringBuilder strBuilder = new StringBuilder();
```
I like your choice of StringBuilder to output everything in one go

starchypotatocode commented on src/main/java/Rizz.java:
```diff
@@ -0,0 +1,154 @@
+
+    private void markTask(int index) {
+            Task task = arrList.get(index - 1);
+            task.markAsDone();
```
I think this should be task.setDone() due to the coding guide requirement of setters being named accordingly?

starchypotatocode commented on src/main/java/Rizz.java:
```diff
@@ -0,0 +1,154 @@
+
+    private void unmarkTask(int index) {
+            Task task = arrList.get(index - 1);
+            task.unmarkAsDone();
```
Same for this: task.setNotDone()?

starchypotatocode reviewed (commented):
Looks good,
some method renaming suggested.

george-yeo commented on src/main/java/Rizz.java:
```diff
@@ -3,47 +3,62 @@
-                "____________________________________________________________\n";
-        System.out.println(this.indent(str));
+        String str = "\t____________________________________________________________\n" +
+                "\tHello! I'm Rizz\n" +
```
Should the chatbot name be extracted out as a final static variable?

george-yeo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        this.isDone = false;
+    }
+
+    public void markAsDone() {
```
Perhaps the setter should be named as setDone(), according to the coding standard?

george-yeo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        this.isDone = true;
+    }
+
+    public void unmarkAsDone() {
```
Similarly, maybe setUndone()?

george-yeo reviewed (commented):
I like that you organized all your logic into discrete methods.
I don't like that there are missing javadocs.

saajidshaik02 committed:
Rizz Level-7: Save

saajidshaik02 committed:
Merge branch 'branch-Level-7'

saajidshaik02 committed:
Rizz Level-8: Dates and Times

saajidshaik02 committed:
Merge branch 'branch-Level-8'

SamuelFoo commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+
+
+    @Override
+    public String export() {
```
I was a little confused about what `export` does initially. Maybe this function could have a more apt naming or there could be JavaDocs explaining what it does?

SamuelFoo commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,19 @@
+        this.tasks = arrList;
+    }
+
+    public String[] exportTasks() {
```
Similar to the point above about naming. I felt that it was unclear what you are exporting the tasks to. Maybe you could name this as `getTasksStorageString`? Sorry, this is a very minor point :sweat_smile:

SamuelFoo commented on src/main/java/Rizz.java:
```diff
@@ -0,0 +1,252 @@
+            System.out.println("\t\t" + task + "\n");
+    }
+
+    private void caseCheck() {
```
Maybe this could be renamed to `parseCommand`?

SamuelFoo commented on src/main/java/Rizz.java:
```diff
@@ -0,0 +1,252 @@
+        System.out.println(str);
+    }
+
+    private void exit() {
```
Maybe this could be named as `getExitString` instead because `exit` implies that the function terminates the programme.

SamuelFoo reviewed (commented):
LGTM! Just some nits about naming. Sorry if I am being too pedantic!

saajidshaik02 committed:
Rizz A-MoreOOP: Use More OOP

saajidshaik02 committed:
Minor Fixes to UI display

saajidshaik02 committed:
Merge tag 'A-MoreOOP'

saajidshaik02 committed:
Rizz A-Packages: Organize into Packages

saajidshaik02 committed:
Merge branch 'A-Packages'

saajidshaik02 committed:
Minor bug fixes

saajidshaik02 committed:
Merge branch 'add-gradle-support'

saajidshaik02 committed:
Rizz A-Gradle: Use Gradle

saajidshaik02 committed:
Rizz A-JUnit: Add JUnit Tests

saajidshaik02 committed:
Merge branch 'branch-JUnit'

saajidshaik02 committed:
Rizz A-JavaDoc: JavaDoc

saajidshaik02 committed:
Merge branch 'branch-A-JavaDoc'

saajidshaik02 committed:
Rizz Level-9: Find

saajidshaik02 committed:
Merge branch 'Level-9'

Fixed # Conflicts:
#	src/main/java/rizz/command/Command.java
#	src/main/java/rizz/task/Task.java

saajidshaik02 committed:
Rizz A-CodingStandard: Follow the Coding Standard

saajidshaik02 committed:
Merge branch 'branch-A-CodingStandard'

Resolved # Conflicts:
#	src/main/java/rizz/command/Command.java
#	src/main/java/rizz/source/Parser.java
#	src/main/java/rizz/source/Rizz.java
#	src/main/java/rizz/source/Storage.java

saajidshaik02 committed:
Rizz A-CheckStyle: Use CheckStyle

saajidshaik02 committed:
Merge branch 'branch-A-CheckStyle'

saajidshaik02 committed:
Rizz Level-10: GUI. Remove Ui Class, Add Main, MainWindow and Launcher Classes to implement JavaFX GUI. Add FXML and CSS files for GUI

saajidshaik02 committed:
Merge branch 'branch-Level-10'

saajidshaik02 committed:
Rizz A-Varargs: Use Varargs. Able to exceute delete, mark and unmark for multiple tasks in 1 input

saajidshaik02 committed:
Merge branch 'branch-A-Varargs'

saajidshaik02 committed:
Rizz A-Assertions: Use Assertions

saajidshaik02 committed:
Rizz A-CodeQuality: Improve Code Quality

Lack of standardisation of variable-naming across files. Includes unconventionally long/too generic/excessively abbreviated functions and variables.

Needs to be fixed for clarity and better understanding across team.

Refactor variable names for clarity: e.g. rename index, by, from and to -> taskIndex,deadlineTimeBy, eventStartTime, eventEndTime.
Added "temp" header to temporary variables such as task and str to indicate the fleeting nature of these created variables (used as intermediatory steps).

Note that these changes have been made according to code-quality conventions mentioned in the article below:
https://se-education.org/guides/conventions/git.html#commit-message-body

saajidshaik02 committed:
Rizz A-Streams: Use Streams

{Current situation}:
The Storage and TaskList classes currently uses traditional loops and StringBuilder. This approach requires manual iteration over lists and manual string concatenation, making the code longer and harder to maintain.

{Why it needs to change}:
Manual looping adds boilerplate code, which reduces readability. Streams can handle the same operations more concisely, making the code easier to understand and maintain.

{What is being done about it}:
Replace the for-loop in loadTasks() with a stream:
Use map() to decode each line into a Task.
Use filter() to remove any null tasks.
Use forEach() to add the decoded tasks to the task list.
Replace the manual StringBuilder loop in saveTasks() with a stream:

{Why it is done that way}:
Streams provide easier way to process collections, reducing boilerplate and improving readability. By leveraging functional methods like map(), filter(), and forEach(), we ensure a more concise and maintainable codebase.

{Any other relevant info}:
NIL

saajidshaik02 committed:
Merge pull request #2 from saajidshaik02/branch-A-Assertions

Rizz A-Assertions: Use Assertions

saajidshaik02 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/rizz/source/TaskList.java

saajidshaik02 committed:
Merge pull request #3 from saajidshaik02/branch-A-CodeQuality

Rizz A-CodeQuality: Improve Code Quality

saajidshaik02 committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/rizz/source/TaskList.java

saajidshaik02 committed:
Merge pull request #4 from saajidshaik02/branch-A-Streams

Rizz A-Streams: Use Streams

saajidshaik02 committed:
Rizz A-CI

Continuous Integration via GitHub Actions. Add gradle.yml

saajidshaik02 committed:
GitHub Actions Error Fix

Resolved duplicate methods in TaskList which caused failure in build during CI Check

saajidshaik02 committed:
GitHub Actions CheckStyle Error

Fixed Checkstyle format

saajidshaik02 committed:
GitHub Actions Error Fix

Changed runtest.bat to fix GitHub Actions Errors

saajidshaik02 committed:
GitHub Actions More Errors Part 2

Remove NIO tests

saajidshaik02 committed:
Merge pull request #5 from saajidshaik02/branch-A-CI

Rizz A-CI

saajidshaik02 committed:
Rizz C-Undo

Implement undo functionality with deep cloning of TaskList

The application needed the ability to undo the most recent task-modifying command.

Initially, we were pushing the current task list directly onto the history stack. This caused issues because subsequent modifications to the task list would affect the stored snapshots in the history, as the same object reference was shared across both.

To fix this, we implemented a deep cloning mechanism for the TaskList, ensuring that each snapshot stored in the history stack is an independent copy of the task list. This allows for proper restoration of previous states when undo is invoked.

We made the following changes:
1. Added a deep copy constructor for TaskList.
2. Updated the `saveSnapshot()` method to push a cloned version of TaskList onto the history stack.
3. Ensured that the undo command restores the task list by popping the last saved snapshot from the history stack.

With this update, undo now works correctly by reverting the task list to its previous state before the most recent change.

saajidshaik02 committed:
Merge branch 'branch-C-Undo'

saajidshaik02 committed:
Rizz A-AiAssisted: Enhance using AI tools

saajidshaik02 committed:
Merge branch 'branch-A-AiAssisted'

saajidshaik02 committed:
Rizz A-Personality: A unique personality

saajidshaik02 committed:
Merge branch 'branch-A-Personality'

saajidshaik02 committed:
Rizz A-UserGuide

saajidshaik02 committed:
Merge branch 'branch-A-UserGuide'

saajidshaik02 committed:
Rename Ui.png file, added files to .gitignore

saajidshaik02 committed:
Minor Fixes to README

saajidshaik02 committed:
Add Authors Acknowledgement Section for User guide
